tests/integrations/litestar/test_litestar.py

```diff
@@ -0,0 +1,326 @@
+
+
+def test_middleware_spans(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+
+    logging_config = LoggingMiddlewareConfig()
+    session_config = ServerSideSessionConfig()
+    rate_limit_config = RateLimitConfig(rate_limit=("hour", 5))
+
+    litestar_app = litestar_app_factory(
+        middleware=[
+            session_config.middleware,
+            logging_config.middleware,
+            rate_limit_config.middleware,
+        ]
+    )
+    events = capture_events()
+
+    client = TestClient(
+        litestar_app, raise_server_exceptions=False, base_url="http://testserver.local"
+    )
+    try:
+        client.get("/message")
+    except Exception:
+        pass
+
+    (_, transaction_event) = events
+
+    expected = ["SessionMiddleware", "LoggingMiddleware", "RateLimitMiddleware"]
+
+    idx = 0
+    for span in transaction_event["spans"]:
+        if span["op"] == "middleware.litestar":
+            assert span["description"] == expected[idx]
+            assert span["tags"]["litestar.middleware_name"] == expected[idx]
+            idx += 1
+
+
+def test_middleware_callback_spans(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+    litestar_app = litestar_app_factory(middleware=[SampleMiddleware])
+    events = capture_events()
+
+    client = TestClient(litestar_app, raise_server_exceptions=False)
+    try:
+        client.get("/message")
+    except Exception:
+        pass
+
+    (_, transaction_event) = events
+
+    expected = [
+        {
+            "op": "middleware.litestar",
+            "description": "SampleMiddleware",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+        {
+            "op": "middleware.litestar.send",
+            "description": "SentryAsgiMiddleware._run_app.<locals>._sentry_wrapped_send",
+            "tags": {"litestar.middleware_name": "SampleMiddleware"},
+        },
+    ]
+    for idx, span in enumerate(transaction_event["spans"]):
+        assert span["op"] == expected[idx]["op"]
+        assert span["description"] == expected[idx]["description"]
+        assert span["tags"] == expected[idx]["tags"]
+
+
+def test_middleware_receive_send(sentry_init, capture_events):
+    sentry_init(
+        traces_sample_rate=1.0,
+        integrations=[LitestarIntegration()],
+    )
+    litestar_app = litestar_app_factory(middleware=[SampleReceiveSendMiddleware])
+
+    client = TestClient(litestar_app, raise_server_exceptions=False)
+    try:
+        # NOTE: the assert statements checking
+        # for correct behaviour are in `SampleReceiveSendMiddleware`!
```

Comment:
Let's please inline the `SampleReceiveSendMiddleware` inside this test. We only use that class here, and by inlining it would be immediately clear what this test is trying to assert. We can likely also remove this comment after doing so.

Reply:
I inlined `SampleReceiveSendMiddleware` but I kept a variation of the comment to point the reader in the right direction.

I did the same for a couple of other classes only used in single test functions, `SampleMiddleware` and `SamplePartialReceiveSendMiddleware`